test/Quake/vector.qke

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                 //
+// All rights reserved.                                                       //
+//                                                                            //
+// This source code and the accompanying materials are made available under   //
+// the terms of the Apache License 2.0 which accompanies this distribution.   //
+// ========================================================================== //
+
+// RUN: cudaq-opt --lower-to-cfg --cse --canonicalize %s | FileCheck %s
+// RUN: cudaq-opt --lower-to-cfg --cse --canonicalize %s | \
+// RUN: cudaq-translate --convert-to=qir | FileCheck --check-prefix=QIR %s
+
+// Blindly assume that %vecvec and %retval are ragged matrices with identical
+// structure and lengths. Convert ints from %vecvec to doubles and copy the
+// values into %retval. Does not create any new std::vectors.
+func.func @vector_vector(%vecvec : !cc.stdvec<!cc.stdvec<i32>>, %retval : !cc.stdvec<!cc.stdvec<f64>>) {
+  %0 = arith.constant 0 : i64
+  %one = arith.constant 1 : i64
+  %vecvecsz = cc.stdvec_size %vecvec : (!cc.stdvec<!cc.stdvec<i32>>) -> i64
+  %resvecs = cc.stdvec_data %retval : (!cc.stdvec<!cc.stdvec<f64>>) -> !cc.ptr<!cc.array<!cc.stdvec<f64> x ?>>
+  cc.loop while ((%arg0 = %0) -> i64) {
+    %c = arith.cmpi ult, %arg0, %vecvecsz : i64
+    cc.condition %c (%arg0 : i64)
+  } do {
+   ^bb0(%arg0 : i64):
+    %vec = cc.stdvec_data %vecvec : (!cc.stdvec<!cc.stdvec<i32>>) -> !cc.ptr<!cc.array<!cc.stdvec<i32> x ?>>
+    %vecpt = cc.compute_ptr %vec[%arg0] : (!cc.ptr<!cc.array<!cc.stdvec<i32> x ?>>, i64) -> !cc.ptr<!cc.stdvec<i32>>
+    %inner = cc.load %vecpt : !cc.ptr<!cc.stdvec<i32>>
+    %ints = cc.stdvec_data %inner : (!cc.stdvec<i32>) -> !cc.ptr<!cc.array<i32 x ?>>
+    %vecsz = cc.stdvec_size %inner : (!cc.stdvec<i32>) -> i64
+    %resultvec = cc.compute_ptr %resvecs[%arg0] : (!cc.ptr<!cc.array<!cc.stdvec<f64> x ?>>, i64) -> !cc.ptr<!cc.stdvec<f64>>
+    %resuvec = cc.load %resultvec : !cc.ptr<!cc.stdvec<f64>>
+    %dubs = cc.stdvec_data %resuvec : (!cc.stdvec<f64>) -> !cc.ptr<!cc.array<f64 x ?>>
+    cc.loop while ((%arg1 = %0) -> i64) {
+      %c = arith.cmpi ult, %arg1, %vecsz : i64
+      cc.condition %c (%arg1 : i64)
+    } do {
+     ^bb0(%arg1 : i64):
+      %ptr = cc.compute_ptr %ints[%arg1] : (!cc.ptr<!cc.array<i32 x ?>>, i64) -> !cc.ptr<i32>
+      %a_ij = cc.load %ptr : !cc.ptr<i32>
+      %dub = cc.cast signed %a_ij : (i32) -> f64
+      %dub_i = cc.compute_ptr %dubs[%arg1] : (!cc.ptr<!cc.array<f64 x ?>>, i64) -> !cc.ptr<f64>
+      cc.store %dub, %dub_i : !cc.ptr<f64>   
+      cc.continue %arg1 : i64
+    } step {
+     ^bb0(%arg1 : i64):
+      %next = arith.addi %arg1, %one : i64
+      cc.continue %next : i64
+    }
+    cc.continue %arg0 : i64
+  } step {
+   ^bb0(%arg0 : i64):
+    %next = arith.addi %arg0, %one : i64
+    cc.continue %next : i64
+  }
+  return
+}
+
+// CHECK-LABEL:   func.func @vector_vector(
```

Comment:
Is there any way for us to test that this LLVM code works when executed? Maybe that can come later when we wire this through the bridge.